measurements/api/measurements.py

```diff
@@ -628,69 +628,73 @@ def coal(colname):
-                        "probe_asn": "AS{}".format(row.probe_asn),
-                        "test_name": row.test_name,
-                        "measurement_start_time": row.measurement_start_time,
-                        "input": row.input,
-                        "anomaly": row.anomaly,
-                        "confirmed": row.confirmed,
-                        "failure": row.failure,
-                        "scores": json.loads(row.scores),
-                    }
-                )
-        except OperationalError as exc:
-            if isinstance(exc.orig, QueryCanceledError):
-                capture_exception(QueryTimeoutError())
-                raise QueryTimeoutError()
-            raise exc
-
-        # For each report_id / input tuple, we want at most one entry from the
-        # traditional pipeline and one from fastpath, merged together
-        results = _merge_results(tmpresults)
-
-        pages = -1
-        count = -1
-        current_page = math.ceil(offset / limit) + 1
-
-        # We got less results than what we expected, we know the count and that we are done
-        if len(results) < limit:
-            count = offset + len(results)
-            pages = math.ceil(count / limit)
-            next_url = None
-        else:
-            # XXX this is too intensive. find a workaround
-            # count_start_time = time.time()
-            # count = q.count()
-            # pages = math.ceil(count / limit)
-            # current_page = math.ceil(offset / limit) + 1
-            # query_time += time.time() - count_start_time
-            next_args = request.args.to_dict()
-            next_args["offset"] = "%s" % (offset + limit)
-            next_args["limit"] = "%s" % limit
-            next_url = urljoin(
+    try:
+        tmpresults = []
+        for row in q:
+            url = urljoin(
                 current_app.config["BASE_URL"],
-                "/api/v1/measurements?%s" % urlencode(next_args),
+                "/api/v1/measurement/%s" % row.measurement_id,
+            )
+            tmpresults.append(
+                {
+                    "measurement_url": url,
+                    "measurement_id": row.measurement_id,
+                    "report_id": row.report_id,
+                    "probe_cc": row.probe_cc,
+                    "probe_asn": "AS{}".format(row.probe_asn),
+                    "test_name": row.test_name,
+                    "measurement_start_time": row.measurement_start_time,
+                    "input": row.input,
+                    "anomaly": row.anomaly,
+                    "confirmed": row.confirmed,
+                    "failure": row.failure,
+                    "scores": json.loads(row.scores),
+                }
             )
+    except OperationalError as exc:
+        if isinstance(exc.orig, QueryCanceledError):
+            # Timeout due to a slow query. Generate metric and do not feed it
+            # to Sentry.
+            raise QueryTimeoutError()
```

Comment:
By raising the exceptions aren't you sending it to sentry?

Reply:
That was my question on Slack: is FlaskIntegration in app.py acting as a catch-all? If so, the previous implementation of timeout handling was unnecessary.